ops/c/src/mpi/ops_mpi_hdf5.cpp

```diff
@@ -2297,3 +2343,620 @@ void ops_write_const_hdf5(char const *name, int dim, char const *type,
   MPI_Comm_free(&OPS_MPI_HDF5_WORLD);
 }
 
+hid_t h5_type(const char *type) {
+  hid_t h5t{0};
+  if (strcmp(type, "double") == 0 || strcmp(type, "real(8)") == 0) {
+    h5t = H5T_NATIVE_DOUBLE;
+  } else if (strcmp(type, "float") == 0 || strcmp(type, "real(4)") == 0 ||
+             strcmp(type, "real") == 0) {
+    h5t = H5T_NATIVE_FLOAT;
+  } else if (strcmp(type, "int") == 0 || strcmp(type, "int(4)") == 0 ||
+             strcmp(type, "integer(4)") == 0) {
+    h5t = H5T_NATIVE_INT;
+  } else if (strcmp(type, "long") == 0) {
+    h5t = H5T_NATIVE_LONG;
+  } else if ((strcmp(type, "long long") == 0) || (strcmp(type, "ll") == 0)) {
+    h5t = H5T_NATIVE_LLONG;
+  } else if (strcmp(type, "short") == 0) {
+    h5t = H5T_NATIVE_SHORT;
+  } else if (strcmp(type, "char") == 0) {
+    h5t = H5T_NATIVE_CHAR;
+
+  } else {
+    OPSException ex(OPS_HDF5_ERROR);
+    ex << "Error: Unknown data type for converting to hdf5 recognised types";
+    throw ex;
+  }
+  return h5t;
+}
+
+void determine_plane_global_range(const ops_dat dat,
+                                  const int cross_section_dir, const int pos,
+                                  int *range) {
+  const int space_dim{dat->block->dims};
+  const sub_dat *sd = OPS_sub_dat_list[dat->index];
+  int *size{new int(space_dim)};
+  for (int d = 0; d < space_dim; d++) {
+    size[d] = sd->gbl_size[d] - (sd->gbl_d_p[d] - sd->gbl_d_m[d]);
+  }
+
+  for (int d = 0; d < dat->block->dims; d++) {
+    range[2 * d] = 0;
+    range[2 * d + 1] = size[d];
+  }
+  range[2 * cross_section_dir + 1] = pos + 1;
+  range[2 * cross_section_dir] = pos;
+  delete size;
+}
+
+void determine_local_range(const ops_dat dat, const int *global_range,
+                           int *local_range) {
+  ops_arg *dat_arg;
+  const int space_dim{dat->block->dims};
+  if (space_dim == 3) {
+    int s3D_000[]{0, 0, 0};
+    ops_stencil S3D_000{ops_decl_stencil(3, 1, s3D_000, "000")};
+    ops_arg arg = ops_arg_dat(dat, dat->dim, S3D_000, dat->type, OPS_READ);
```

Comment:
this will go out of scope after the next line. Need to pull out above if

Reply:
that is just a pointer. it will point to "arg" which goes out of scope at the end of the if block

Reply:
Now slighly changed to anothe away, possbily more concise.